### PR TITLE
va-on-this-page hide link to h2 without id

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Chromatic

https://37776-va-otp-hash--60f9b557105290003b387cd5.chromatic.com

## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/37776

## Testing done
E2E, Storybook

## Screenshots
Screen Shot 2022-03-31 at 17 47 54

## Acceptance criteria
- [x] `va-on-this-page` only displays h2s containing ids

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by andresriveratoro** [comment]
This web component's responsibility is to create anchors that scroll the user's page to the `h2` element (inside an article) when clicked. Should it be our responsibility to create an ESLint rule to enforce ids on h2s on `vets-website`?

**Review by pdavies88 — APPROVED** [review]

**Comment by pdavies88** [comment]
> This web component's responsibility is to create anchors that scroll the user's page to the `h2` element (inside an article) when clicked. Should it be our responsibility to create an ESLint rule to enforce ids on h2s inside `article` on `vets-website`?

I think from our convo during the meeting today that the consoleDevError should be enough but @bkjohnson correct me if I am wrong on that.

**Review by bkjohnson — COMMENTED** [review]
Should we have a story showing this behavior? I'm hesitant because I don't know if it's something we want to encourage/advertise.

As far as the ESLint rule goes, I think that we can release this feature without that and if folks decide that that's something that needs to be in place we can add it later.

**Comment by pdavies88** [comment]
> Should we have a story showing this behavior? I'm hesitant because I don't know if it's something we want to encourage/advertise.
> 
> As far as the ESLint rule goes, I think that we can release this feature without that and if folks decide that that's something that needs to be in place we can add it later.

@bkjohnson @andresriveratoro personally I am in the camp of not showing this behavior since really this shouldn't be the way it is used and is more of a safeguard if it occurs

**Review by bkjohnson — COMMENTED** [review]
I'm fine with not having a story - just one small comment.

File: packages/storybook/stories/va-on-this-page.stories.jsx [hunk 1]
```diff
@@ -2,6 +2,14 @@ import React from 'react';
 
 export default {
   title: 'Components/va-on-this-page',
+  parameters: {
+    docs: {
+      description: {
+        component: `This component will render links based on the content around it. It scans the document for any h2 elements
+        that contain an **id** inside of an <article> and will create a list of links to the headings.`,
+      },
```
[the remark this answers is not in the document]
Comment (andresriveratoro) [reply packages/storybook/stories/va-on-this-page.stories.jsx]: That is way better. Updated.

**Pull request merged**